src/include/common/re2_regex.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <stdexcept>
+#include "re2.h"
+
+namespace kuzu::common {
+enum struct RegexOptions : uint8_t { NONE, CASE_INSENSITIVE };
+
+class Regex {
+public:
+    Regex(const std::string &pattern, RegexOptions options = RegexOptions::NONE);
+    const kuzu::regex::RE2 &GetRegex() const {
+        return *regex;
+    }
+
+private:
+    std::shared_ptr<kuzu::regex::RE2> regex;
+};
+
+struct GroupMatch {
+    std::string text;
+    uint32_t position;
+
+    const std::string &str() const {
+        return text;
+    }
+    operator std::string() const {
+        return text;
+    }
+};
+
+struct Match {
+    std::vector<GroupMatch> groups;
+
+    GroupMatch &GetGroup(uint64_t index) {
+        if (index >= groups.size()) {
+            throw std::runtime_error("RE2: Match index is out of range");
+        }
+        return groups[index];
+    }
+
+    std::string str(uint64_t index) {
```

Comment:
Our convention on class method naming is that:
If a method is used to retrive a field from a class, then we name it as: `getXxx`.
E.g. better to rename `position` => `getPosition`.